salt/orch/update.sls

```diff
@@ -0,0 +1,117 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+# TODO: When we update to 2016.11.4 uncomment this
+# update_grains:
+#   salt.function:
+#     - tgt: '*'
+#     - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+
+update_modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+
+# Get list of masters needing reboot
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master and G@tx_update_reboot_needed:true', fun='network.ip_addrs', tgt_type='compound') %}
+
+{%- for master_id in masters.keys() %}
+
+{{ master_id }}-clean-shutdown:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kubernetes-master.stop
+      - flannel.stop
+      - etcd.stop
+
+# Reboot the node
+{{ master_id }}-reboot:
+  salt.function:
+    - tgt: {{ master_id }}
+    - name: cmd.run
+    - arg:
+      - rebootmgrctl reboot now
+
+# Wait for it to start again
+{{ master_id }}-wait-for-start:
+  salt.wait_for_event:
+    - name: salt/minion/*/start
+    - id_list:
+      - {{ master_id }}
+
+# Start services
+{{ master_id }}-start-services:
+  salt.state:
+    - tgt: {{ master_id }}
+    - highstate: True
+
+{{ master_id }}-update-grain:
```

Comment:
It could easily make a difference to the transactional-update code that updates this file and sets the flag to true! better to have it always true/false, rather than true or "falsy"